### PR TITLE
dlgClimsoft

### PR DESCRIPTION
**Pull request by stevekogo**

### CODE REVIEW

**Comment by stevekogo** [comment]
This is not ready for review but it does importation.

**Comment by stevekogo** [comment]
@africanmathsinitiative/developers  This is ready for review.

**Comment by stevekogo** [comment]
@rdstern, @volloholic , @dannyparsons  I have made significant progress on importing climsoft data. I will be happy to demonstrate the status of this then you can give your views/suggestions.

File: instat/dlgClimSoft.vb [hunk 1]
```diff
@@ -27,32 +27,57 @@ Public Class dlgClimSoft
...
         ucrReceiverMultipleElements.Selector = ucrSelectorForClimSoft
         ucrReceiverMultipleElements.SetItemType("database_variables")
         ucrReceiverMultipleElements.strDatabaseQuery = "SELECT obselement.elementName FROM obselement,observationfinal WHERE obselement.elementId=observationfinal.describedBy AND observationfinal.recordedFrom in (10202200,10306100) GROUP BY observationfinal.describedBy;"
+
         ucrChkObservationData.SetText("Observation Data")
+        ucrChkObservationData.SetParameter(New RParameter("include_observation_data", 2))
+        ucrChkObservationData.SetValuesCheckedAndUnchecked("TRUE", "FALSE")
+
+        ucrInputStartDate.SetParameter(New RParameter("start_date", 3))
+        ucrInputEndDate.SetParameter(New RParameter("end_date", 4))
```
Comment (lilyclements) [inline instat/dlgClimSoft.vb]: Do these ucrInput's need a SetValidationTypeAsRVariable line?

File: instat/dlgClimSoft.vb [hunk 2]
```diff
@@ -27,32 +27,57 @@ Public Class dlgClimSoft
...
     End Sub
 
     Private Sub TestOKEnabled()
+        If (Not ucrReceiverMultipleStations.IsEmpty() AndAlso Not ucrChkObservationData.Checked) OrElse (Not ucrReceiverMultipleStations.IsEmpty() AndAlso Not ucrReceiverMultipleElements.IsEmpty() AndAlso ucrChkObservationData.Checked) Then
```
Comment (lilyclements) [inline instat/dlgClimSoft.vb]: How about if the start and end dates are empty too? Or are they not necessary?

File: instat/dlgClimSoft.vb [hunk 3]
```diff
@@ -79,13 +107,20 @@ Public Class dlgClimSoft
...
             ucrReceiverMultipleElements.strDatabaseQuery = "SELECT obselement.elementName FROM obselement,observationfinal WHERE obselement.elementId=observationfinal.describedBy GROUP BY observationfinal.describedBy;"
         Else
-            ucrReceiverMultipleElements.strDatabaseQuery = "SELECT obselement.elementName FROM obselement,observationfinal WHERE obselement.elementId=observationfinal.describedBy AND observationfinal.recordedFrom IN (" & String.Join(",", ucrReceiverMultipleStations.GetVariableNamesList) & ") GROUP BY observationfinal.describedBy;"
+            ucrReceiverMultipleElements.strDatabaseQuery = "SELECT obselement.elementName FROM obselement,observationfinal WHERE obselement.elementId=observationfinal.describedBy AND observationfinal.recordedFrom IN (" & String.Join(",", ucrReceiverMultipleStations.GetVariableNamesList(bWithQuotes:=False)) & ") GROUP BY observationfinal.describedBy;"
         End If
+        TestOKEnabled()
+    End Sub
+
+    Private Sub ucrControlsContents_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrReceiverMultipleElements.ControlValueChanged, ucrChkObservationData.ControlContentsChanged
```
Comment (lilyclements) [inline instat/dlgClimSoft.vb]: There are items in your TestOK sub which you have not mentioned here

**Comment by rdstern** [comment]
Just loaded the latest R-Instat and started by the climsoft dialogue.  It gave an error and threw me out immediately.  Here it is:

---------------------------
Error running R command(s)
---------------------------
Error in if (self$has_database_connection) { : 
  argument is not interpretable as logical


The error occurred in attempting to run the following R command(s):

InstatDataObject$get_database_variable_names(exclude=list(Is_Hidden=TRUE), as_list=TRUE, query="SELECT stationId FROM station;")
---------------------------
OK   
---------------------------

**Comment by dannyparsons** [comment]
I believe this will be fixed by this pull request, which isn't merged yet.

**Comment by stevekogo** [comment]
@africanmathsinitiative/developers Someone should have a look at this.

**Pull request merged**